### PR TITLE
Nojira | @ericbakenhus | Remove draft versions of published events

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Removes the draft version of any events that are published with changes (will appear in both published and draft requests)
- Splits out `isPublished` to prevent it from appearing in other systems

# Review By (Date)
- 🤷 

# Criticality
- Medium(?)

# Review Tasks

## Setup tasks and/or behavior to test

1. Look at code
2. Maybe ping the deploy preview background function with Postman and watch the logs (feel free to set `run` to `test` in vaut if you don't want it to make changes in the dev SB space; doesn't really matter either way though)
    - https://deploy-preview-981--stanford-alumni.netlify.app/webhook/sb/event-import?secret=[secret-here]

### CODE REVIEW

File: netlify/functions/event-import-background.mts [hunk 1]
```diff
@@ -107,40 +107,62 @@ export default async (req: Request) => {
...
+    const sbPublishedEvents = await storyblokContent.getAll('cdn/stories', {
+      starts_with: 'events/sync/',
       content_type: 'synchronizedEvent',
       version: 'published',
       per_page: 100,
     }) ?? [];
-    const sbUnpublishedEvents = await storyblokContent.getAll('cdn/stories', { 
-      starts_with: 'events/sync/', 
-      content_type: 'synchronizedEvent', 
+    const sbDraftEvents = await storyblokContent.getAll('cdn/stories', {
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:116]: Make the name more obvious.

File: netlify/functions/event-import-background.mts [hunk 2]
```diff
@@ -107,40 +107,62 @@ export default async (req: Request) => {
...
+    const oldArchivedEvents = allOldArchivedEvents.filter((s) => {
+      if (seenArchivedEventIds.has(s.id)) {
+        console.log('Duplicate archived event found (unpublished changes): ', s.id);
+        return false;
+      }
+
+      seenArchivedEventIds.add(s.id);
+      return true;
+    });
+
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:165]: All the duplicate filtering.

File: netlify/functions/event-import-background.mts [hunk 3]
```diff
@@ -169,14 +191,14 @@ export default async (req: Request) => {
...
+    sbEvents.forEach(({ isPublished, ...story }) => {
       const id = story.content.externalId;
 
       if (id) {
         const existing = syncedEvents.get(id);
         const value = existing
-          ? { ...existing, storyblok: story }
-          : { google: undefined, storyblok: story }
+          ? { ...existing, storyblok: story, isPublished }
+          : { google: undefined, storyblok: story, isPublished }
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:201]: Pull our `isPublished` out of the story object and pass it along separately.

File: netlify/functions/event-import-background.mts [hunk 4]
```diff
@@ -224,21 +246,21 @@ export default async (req: Request) => {
       }
     }
 
-    for (const [id, {google, storyblok}] of syncedEvents) {
+    for (const [id, {google, storyblok, isPublished}] of syncedEvents) {
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:249]: `isPublished` is a separate key now.

File: netlify/functions/event-import-background.mts [hunk 5]
```diff
@@ -283,16 +305,16 @@ export default async (req: Request) => {
                 ...combinedStory,
                 parent_id: eventFolderId,
               },
-              publish: storyblok.isPublished ? 1 : 0, // Don't re-publish manually unpublished events
+              publish: isPublished ? 1 : 0, // Don't re-publish manually unpublished events
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:308]: Use new `isPublished`.

File: netlify/functions/event-import-background.mts [hunk 6]
```diff
@@ -333,21 +355,23 @@ export default async (req: Request) => {
...
+                story: {
                   ...storyblok,
                   parent_id: eventArchiveFolderId,
                 },
               });
               await delay();
             }
             console.log('Done!');
-          } else if (storyblok.isPublished) {
+          } else if (isPublished) {
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:366]: Use new `isPublished`.

File: netlify/functions/event-import-background.mts [hunk 7]
```diff
@@ -333,21 +355,23 @@ export default async (req: Request) => {
...
             console.log('Unpublishing...');
             if (run) {
               await storyblokManagement.get(`/spaces/${spaceId}/stories/${storyblok.id}/unpublish`);
               await delay();
             }
             console.log('Unpublished!');
+          } else {
+            console.log('Only in SB and not published. No changes needed.');
```
Comment (ericbakenhus) [inline netlify/functions/event-import-background.mts:374]: Extra logging when we don't need to do anything when we run into an unpublished, unexpired event that's been deleted from the Google sheets.

**Review by sherakama — APPROVED** [review]
Tank u.

**Comment by ericbakenhus** [comment]
tank-cat

**Pull request merged**